tools/pubsys/src/repo.rs

```diff
@@ -286,6 +288,61 @@ fn repo_urls<'a>(
     Ok(None)
 }
 
+/// If the infra config has a repo section defined for the given repo, and it has a metadata base URL and
+/// a targets URL defined, compose and returns the set of metadata URLs and their target URL using the specified arches
+/// and variants, otherwise return None.
+fn compose_repo_urls<'a>(
+    repo_key: String,
+    infra_config: &'a InfraConfig,
+) -> Result<Option<(Vec<Url>, &'a Url)>> {
+    let repo_config = infra_config
+        .repo
+        .as_ref()
+        .context(error::MissingConfig {
+            missing: "repo section",
+        })?
+        .get(&repo_key)
+        .context(error::MissingConfig {
+            missing: format!("definition for repo {}", &repo_key),
+        })?;
+
+    // Make sure all fields are set and then compose the set of metadata URLs
+    if let (Some(metadata_base_url), Some(targets_url), Some(arches), Some(variants)) = (
```

Comment:
Is there any reason not to throw an error if one of these isn't set? You can't compose a valid repo URL.

Reply:
I do throw an error in `validate_repo::run()`; if this function returns `Ok(None)` then it means I can't compose any repo URLs of the format `${metadata_base_url}${base_slash}${variant}/${arch}`

Reply:
Oh I guess I could just return a result instead of an option in that case. Silly me